### PR TITLE
DDP-5455 fix read-only calculation in Patch Answers

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Fixes DDP-5455.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceDto.java [hunk 1]
```diff
@@ -95,10 +95,6 @@ public Long getFirstCompletedAt() {
         return firstCompletedAt;
     }
 
-    public boolean isReadonly() {
-        return Optional.ofNullable(isReadonly).orElse(false);
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceDto.java:99]: I removed this method so we don't default read-only to `false` and cause another issue with bypassing check.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java [hunk 2]
```diff
@@ -169,9 +169,8 @@ public Object handle(Request request, Response response) {
             }
 
             if (ActivityInstanceUtil.isReadonly(formActivityDef.getEditTimeoutSec(), instanceDto.getCreatedAtMillis(),
-                    instanceDto.getStatusType().name(), formActivityDef.isWriteOnce(), instanceDto.isReadonly())) {
-                String msg = "Activity instance with GUID " + instanceGuid
-                        + " is read-only, cannot submit answer(s) for it";
+                    instanceDto.getStatusType().name(), formActivityDef.isWriteOnce(), instanceDto.getReadonly())) {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/route/PatchFormAnswersRoute.java:172]: `isReadonly()` -> `getReadonly()` is the fix.

**Review by MocanaAtBroad — DISMISSED** [review]

**Comment by MocanaAtBroad** [comment]
You sure you want to allow getReadOnly() to be a null value?  Would it make sense to have a default value of false in the database? 3 possible values for a boolean (even a Boolean) is a little strange.

**Comment by yufengwng** [comment]
@MocanaAtBroad I recall that it was a deliberate choice. The read-only on an activity instance acts as a "master" switch. If we want to override everything else and mark an instance as NOT read-only, we'll set it to `false`. So with that, we need to allow `null` or else every instance will bypass read-only check...

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**